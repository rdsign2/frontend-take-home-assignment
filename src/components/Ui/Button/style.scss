@import '~/assets/styles/configs.scss';

.Button {
  background: $color_primary;
  border-radius: 100rem;
  margin: 0 auto;
  padding: 2rem 0;
  width: 100%;

  &:hover {
    background: $color_primary_soft;
    box-shadow: 0 1rem 3rem rgba($color_primary, 0.3);
    transform: scale(1.02);
  }

  &:active {
    transform: scale(1);
  }

  &:disabled {
    background: transparent;
    color: $color_light_hard;
    cursor: default;

    &:hover {
      background: inherit;
      box-shadow: none;
      transform: none;
    }
  }
}
