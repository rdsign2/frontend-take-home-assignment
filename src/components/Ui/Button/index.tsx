import * as React from 'react';
import './style.scss';

interface Props {
  enable?: boolean;
}

const Button: React.FC<Props> = props => (
  <button className="Button tp" disabled={!props.enable}>
    Confirm
  </button>
);

export default Button;
