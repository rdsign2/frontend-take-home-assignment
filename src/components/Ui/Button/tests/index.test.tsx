import * as React from 'react';
import { shallow } from 'enzyme';
import Button from '../index';

const component = shallow(<Button enable={true} />);

describe('DatePicker', () => {
  it('renders', () => {
    expect(component.exists()).toEqual(true);
  });

  it(`Check if the button is active as enable prop is true`, () => {
    expect(component.prop('disabled')).toBe(false);
  });

  it('snapshotShot', () => {
    expect(component.getElements()).toMatchSnapshot();
  });
});
