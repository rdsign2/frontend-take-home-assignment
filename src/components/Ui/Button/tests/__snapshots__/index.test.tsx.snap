// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker snapshotShot 1`] = `
Array [
  <button
    className="Button tp"
    disabled={false}
  >
    Confirm
  </button>,
]
`;
